src/NServiceBus.Metrics.AcceptanceTests/Tests/QueueLength/When_publishing_message.cs

```diff
@@ -38,26 +37,45 @@ public class When_publishing_message : NServiceBusAcceptanceTest
                     await session.Subscribe<TestEventMessage1>();
                     await session.Subscribe<TestEventMessage2>();
                 }))
-                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2)
+                .WithEndpoint<MonitoringSpy>()
+                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2 && SequencesReported(c))
                 .Run()
                 .ConfigureAwait(false);
 
-            var sessionIds = new[] { AssertHeaders(context.Headers1), AssertHeaders(context.Headers2) };
-
-            var data = JObject.Parse(context.Data);
-            var counters = (JArray)data["Counters"];
-            var counterTokens = counters.Where(c => c.Value<string>("Name").StartsWith("Sent sequence for"));
+            SequencesReported(context);
```

Comment:
> This call is included in the .Done condition above so doesn't that mean that all the checks have been done by this point?

Removed the unneeded call at the end of the test.

> Plus, all the asserts in it are wrapped in a try-catch so this call doesn't really do anything.

Try-catch is needed to return true/false from the method instead of throwing. The reason why this condition has to be checked in `.Done` is that there is no reliable way to know when the metrics report should contain sequence data unless the reporting is broken. Is is caused by the fact that report is sent async to processing messages. In effect this test will wait until sequences are reported or time-out and fail.